packages/analytics-js-integrations/src/integrations/Braze/browser.js

```diff
@@ -75,7 +75,17 @@ class Braze {
   }
 
   isReady() {
-    return this.isLoaded();
+    if (this.isLoaded()) {
+      try {
+        const anonymousId = this.analytics.getAnonymousId();
+        window.braze.getUser().addAlias(anonymousId, 'rudder_id');
+        return true;
+      } catch (error) {
+        logger.error(`Error in isReady - ${stringifyWithoutCircularV1(error, true)}`);
+        return false;
```

Comment:
If `false` is returned, the core SDK will repeatedly attempt the `isReady` check, leading to multiple error statements on the console.

Do we want to handle this better?

Reply:
Any suggestion?

Reply:
You can maintain the state of this operation in the class to log just once and then log again if it gets successful.

Moreover, the API returns a [boolean](https://js.appboycdn.com/web-sdk/latest/doc/classes/braze.user.html#addalias) which we're ignoring. Apart from the error handling, we should also use the return value to maintain the state.

Also, do we want to consider the destination as not ready if alias couldn't be set?

Reply:
Without setting the `alias` we cannot set the `anonymousId` to the profile which is required for
- Keeping these profiles consistent like cloud mode
- Stitch profiles for hybrid mode

Reply:
I didn't get state part on every page refresh this entire class script will get executed how will be the state maintained in that case? Or are talking of using some session storage kind of state?

Reply:
I have updated the code can you check once